app/src/components/Button.tsx

```diff
@@ -1,11 +1,26 @@
 import type { PropsWithChildren } from 'react'
 
 interface ButtonProps extends PropsWithChildren {
-  handleClick: () => void
+  handleClick?: () => void
+  styles: 'general' | 'submit' | 'delete'
```

Comment:
Could you extract this into an array (`BUTTON_STYLES` for example).
I think then you can use it in the interface as `typeof BUTTON_STYLES`, but I'm not 100% sure.